src/query/procedure/module.cpp

```diff
@@ -300,6 +304,89 @@ void RegisterMgTransformations(const std::map<std::string, std::unique_ptr<Modul
+    // We expect modules to be sorted by name.
+    const std::map<std::string, std::unique_ptr<Module>, std::less<>> *all_modules, BuiltinModule *module) {
+  auto functions_cb = [all_modules](mgp_list * /*args*/, mgp_graph * /*graph*/, mgp_result *result,
+                                    mgp_memory *memory) {
+    // Iterating over all_modules assumes that the standard mechanism of custom
+    // procedure invocations takes the ModuleRegistry::lock_ with READ access.
+    // For details on how the invocation is done, take a look at the
+    // CallProcedureCursor::Pull implementation.
+    for (const auto &[module_name, module] : *all_modules) {
+      // Return the results in sorted order by module and by procedure.
+      static_assert(std::is_same_v<decltype(module->Functions()), const std::map<std::string, mgp_func, std::less<>> *>,
+                    "Expected module procedures to be sorted by name");
+
+      const auto path = module->Path();
+      const auto path_string = GetPathString(path);
+      const auto is_editable = IsFileEditable(path);
+
+      for (const auto &[proc_name, proc] : *module->Functions()) {
+        mgp_result_record *record{nullptr};
+        {
+          const auto success = TryOrSetError([&] { return mgp_result_new_record(result, &record); }, result);
+          if (!success) {
+            return;
+          }
+        }
+
+        const auto path_value = GetStringValueOrSetError(path_string.c_str(), memory, result);
+        if (!path_value) {
+          return;
+        }
+
+        MgpUniquePtr<mgp_value> is_editable_value{nullptr, mgp_value_destroy};
+        {
+          const auto success = TryOrSetError(
+              [&] { return CreateMgpObject(is_editable_value, mgp_value_make_bool, is_editable, memory); }, result);
+          if (!success) {
+            return;
+          }
+        }
+
+        utils::pmr::string full_name(module_name, memory->impl);
+        full_name.append(1, '.');
+        full_name.append(proc_name);
+        const auto name_value = GetStringValueOrSetError(full_name.c_str(), memory, result);
+        if (!name_value) {
+          return;
+        }
+
+        std::stringstream ss;
+        ss << module_name << ".";
+        PrintFuncSignature(proc, &ss);
+        const auto signature = ss.str();
+        const auto signature_value = GetStringValueOrSetError(signature.c_str(), memory, result);
+        if (!signature_value) {
+          return;
+        }
+
+        if (!InsertResultOrSetError(result, record, "name", name_value.get())) {
+          return;
+        }
+
+        if (!InsertResultOrSetError(result, record, "signature", signature_value.get())) {
+          return;
+        }
+
+        if (!InsertResultOrSetError(result, record, "path", path_value.get())) {
+          return;
+        }
+
+        if (!InsertResultOrSetError(result, record, "is_editable", is_editable_value.get())) {
+          return;
+        }
+      }
+    }
```

Comment:
I think this could be unified with the procedures ones with some template magic:
```

template <typename T>
const std::map<std::string, T, std::less<>> * GetCallables(Module& module) {
   if constexpr (std::is_same_v<TCall, mgp_proc>) {
      return module.Procedures();
    } else if constexpr (std::is_same_v<TCall, mgp_func>) {
      return module.Functions();
    } 
}
```

If you need help with that, let me know.